### PR TITLE
fix(oauth): google oauth init fails due to unknown db user 

### PR DESCRIPTION
**Pull request by ig-ra**
google oauth fails due to recent changes when it tries to save vars via vars service.
This happens since there is no db user in context (nether via token or session).
Solution - access oauth via authMux, thus passing via auth middleware and adding user to the context

### CODE REVIEW

**Comment by itayd** [comment]
are you sure re authMux? i don't think that oauth redirect have the token we need. @daabr ?
did you test it? i mean manually, end to end?

**Comment by ig-ra** [comment]
> are you sure re authMux? i don't think that oauth redirect have the token we need. @daabr ? did you test it? i mean manually, end to end?

1. Yes, I tested this - previously it failed on google auth. With this fix it's working
2. since we redirected via browser we have our cookie, and thus the user. That's the reason to pass it through authMux - authMiddleware will extract the authenticated user from cookie and put it in the context

File: integrations/google/server.go
Comment (daabr) [inline integrations/google/server.go:27]: Please add a function documentation comment for `Start` to explain that it's currently the only integration that uses `muxAuth`, what `muxAuth` is (in comparison to the usual `muxNoAuth`), and why it's needed at all. In other words, the context and description of this PR as a comment here, to help future maintainers of this code.

**Review by daabr — APPROVED** [review]
Approved, but before merging please add a Go function comment to `Start` - see my PR comment there.

**Pull request merged**
